### PR TITLE
Add atomsk

### PR DESCRIPTION
**Pull request by jan-janssen**
Atomsk - The Swiss-army knife of atomic simulations

### CODE REVIEW

**Comment by jan-janssen** [comment]
There seems to be an issue with circleci at least I can not access the logs right now.

**Comment by jan-janssen** [comment]
I tried the new compiler format, but that did not work for me https://github.com/conda-forge/staged-recipes/pull/5920/commits/a61034d571b67566188c938e4cc46b3fe9fd87e9

File: recipes/atomsk/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  url: https://github.com/pierrehirel/{{ name }}/archive/Beta-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  skip: True  # [not linux]
+
+requirements:
+  build:
+    - gcc
```
Comment (marcelotrevisani) [inline recipes/atomsk/meta.yaml]: Could you please change this part to the new conda build 3? like
```
- {{ compiler('c') }}
```

File: recipes/atomsk/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    - make 
+    - lapack
+  host:
+    - lapack
+    - libgcc
+  run:
+    - lapack
+    - libgcc
```
Comment (marcelotrevisani) [inline recipes/atomsk/meta.yaml]: after you add the syntax for conda build 3, could you also remove the ``libgcc`` from ``run`` and ``host`` ?

File: recipes/atomsk/build.sh [hunk 3]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+export LAPACK=$PREFIX/lib/liblapack.so
```
Comment (marcelotrevisani) [inline recipes/atomsk/build.sh]: Please, change this part to:
```
export LAPACK=$PREFIX/lib/liblapack${SHLIB_EXT}
```

File: recipes/atomsk/meta.yaml (hunk 1 above)
Comment (jan-janssen) [reply recipes/atomsk/meta.yaml]: That’s what I tried in 
https://github.com/conda-forge/staged-recipes/pull/5920/commits/bde69fcb57dbc06432fe63a09f1235e23fd0a2df But somehow it did not work.

Comment (marcelotrevisani) [reply recipes/atomsk/meta.yaml]: Could you please send me the link of the CI with the errors?

**Comment by marcelotrevisani** [comment]
I will try just one thing in your PR

File: recipes/atomsk/meta.yaml (hunk 1 above)
Comment (jan-janssen) [reply recipes/atomsk/meta.yaml]: https://circleci.com/gh/conda-forge/staged-recipes/27636

**Comment by marcelotrevisani** [comment]
I'm not so sure but I think you have to add ``blas``, I will ping @ocefpaf if he has a hint here

File: recipes/atomsk/meta.yaml (hunk 2 above)
Comment (ocefpaf) [reply recipes/atomsk/meta.yaml]: We need to add a new lint for `gcc`/`libgcc` that points to the docs on how to use the compiler jinja syntax.

**Comment by jan-janssen** [comment]
If I understand the error message correctly then ld complains when I use a different compiler than the one used to compile lapack, which still follows the old compiler style - https://github.com/conda-forge/lapack-feedstock/blob/master/recipe/meta.yaml . Meaning I can only use the new conda 3 style when lapack is using the new conda 3 style, but this most likely breaks a lot of other packages.

**Comment by jan-janssen** [comment]
Is there any chance we can merge this in the old - working format - and update when lapack is updated ?

File: recipes/atomsk/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  number: 0
+  skip: True  # [not linux]
+
+requirements:
+  build:
+    - gcc
+    - make 
+    - lapack
+  host:
+    - lapack
```
Comment (isuruf) [inline recipes/atomsk/meta.yaml]: Don't use lapack. Use openblas package which comes with lapack

**Comment by jan-janssen** [comment]
@isuruf Thanks a lot, I was not aware of this. I updated the recipe to match the conda 3 style and everything works fine. All tests are passed - just waiting for the appveyor tests - but windows is anyway skipped.

**Comment by jan-janssen** [comment]
@marcelotrevisani I am not sure who is responsible for reviewing this pull request, but maybe you can have a look again.

**Comment by marcelotrevisani** [comment]
Looks good to me! I'm going to merge this!

Thanks to put this in ``conda-forge`` !  :)

File: recipes/atomsk/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+build:
+  number: 0
+  skip: True  # [not linux]
+
+requirements:
+  build:
+    - {{ compiler('fortran') }}
+    - {{ compiler('c') }}
+    - make 
+    - openblas
```
Comment (isuruf) [inline recipes/atomsk/meta.yaml]: Remove openblas here. It should be needed only in host and run

**Pull request merged**